drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/popovers/definitions/BaseColumnDefinitionBuilder.java

```diff
@@ -0,0 +1,109 @@
+
+    private static final String WHEN = "when\n";
+    private static final String THEN = "then\n";
+    private static final String END = "end\n";
+
+    private Caller<GuidedDecisionTableEditorService> service;
+
+    public BaseColumnDefinitionBuilder( final Caller<GuidedDecisionTableEditorService> service ) {
+        this.service = service;
+    }
+
+    protected void generateDefinitionOnServer( final GuidedDecisionTable52 partialModel,
+                                               final Path path,
+                                               final Callback<String> afterGenerationCallback ) {
+        service.call( ( String source ) -> afterGenerationCallback.callback( source ) ).toSource( path,
+                                                                                                  partialModel );
+
+    }
+
+    protected List<DTCellValue52> makeRowData( final ColumnUtilities columnUtilities,
+                                               final BaseColumn column ) {
+        final List<DTCellValue52> row = new ArrayList<>();
+        row.add( new DTCellValue52( 1 ) );
+        row.add( new DTCellValue52( "desc" ) );
+
+        final DataType.DataTypes dataType = columnUtilities.getDataType( column );
+        row.add( makeCell( dataType ) );
+        return row;
+    }
+
+    protected DTCellValue52 makeCell( final DataType.DataTypes dataType ) {
+        switch ( dataType ) {
+            case NUMERIC:
+            case NUMERIC_BIGDECIMAL:
+            case NUMERIC_BIGINTEGER:
+            case NUMERIC_BYTE:
+            case NUMERIC_DOUBLE:
+            case NUMERIC_FLOAT:
+            case NUMERIC_INTEGER:
+            case NUMERIC_LONG:
+            case NUMERIC_SHORT:
+                return new DTCellValue52( 1 );
+            case DATE:
+                return new DTCellValue52( new Date() );
+            case BOOLEAN:
+                return new DTCellValue52( true );
+        }
+        return new DTCellValue52( "x" );
+    }
+
+    protected String getLHS( final String drl ) {
+        final String _drl = drl.toLowerCase();
+        final int start = _drl.indexOf( WHEN );
+        final int end = _drl.indexOf( THEN );
+
+        if ( start < 0 || end < 0 || end < start ) {
+            return drl;
+        }
+        return drl.substring( start + THEN.length(),
+                              end ).trim().replaceAll( "\t",
+                                                       "" ).replaceAll( "\n", "<br/>" );
+    }
+
+    protected String getRHS( final String drl ) {
+        final String _drl = drl.toLowerCase();
+        final int start = _drl.indexOf( THEN );
+        final int end = _drl.indexOf( END );
+        if ( start < 0 || end < 0 || end < start ) {
+            return drl;
+        }
+        return drl.substring( start + THEN.length(),
+                              end ).trim().replaceAll( "\t",
+                                                       "" ).replaceAll( "\n", "<br/>" );
+    }
+
```

Comment:
These methods are very similar ;). It would make sense to refactor into one, imo.